src/commands/authn/authnCommand.ts

```diff
@@ -0,0 +1,21 @@
+import { AuthnInterface } from "../../interfaces/authn";
+
+export default class AuthnCommand implements AuthnInterface {
+    readonly username: string;
+    readonly password: string;
+
+    constructor(payload: AuthnInterface) {
+        const { username, password } = payload;
```

Comment:
no need for the constructor to accept a typed parameter. It can just take anything that looks like a JSON, even `any` I think, and then it will validate that the parameter has the right fields and assign it a type/interface when returning. 
Also payload isn't the right name here, the constructor doesn't know where the JSON comes from, whether it's a nats payload or someone's object literal. I'd suggest a generic parameter: `data: any`